src/environment.cpp

```diff
@@ -275,12 +283,15 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 		state->m_previous_node = state->m_iterator.m_current_node_pos;
 		state->m_iterator.next();
 	}
-	// Return empty PointedThing if nothing left on the ray
+	
+	// Return empty PointedThing if nothing left on the ray or it is blocking pointable
 	if (state->m_found.empty()) {
-		result->type = POINTEDTHING_NOTHING;
+		result_p->type = POINTEDTHING_NOTHING;
 	} else {
-		*result = state->m_found.top();
-		state->m_found.pop();
+		*result_p = state->m_found.top();
```

Comment:
The `state->m_found.pop();` is required. Or else lua raycasts won't be able to iterate through.

Reply:
fixed

Reply:
fixed